functions/src/apple.ts

```diff
@@ -16,14 +16,13 @@ const signerKeyPassphrase =
 const teamIdentifier = process.env.APPLE_WALLET_CERTS_TEAM_ID;
 
 // apple wallet ticket
-export const createTicket = onCall(async (_, res) => {
-	const context = res as Context;
-	if (!context?.auth) {
+export const createTicket = onCallCustom(async (req) => {
+	if (!req.auth) {
```

Comment:
Couple things:
- `onCallCustom` is a small wrapper over firebase's `onCall` with safer typing (uses `unknown` instead of `any`) and handles zod validation errors
- In firebase functions Gen 2, the handler is no longer `(data, context) => {...}` but `(req, res) => {...}` and `auth` is on the `req` object